tests/bgp/test_ipv6_bgp_scale.py

```diff
@@ -0,0 +1,376 @@
+        ternimated.set()
+        traffic_thread.join()
+        end_time = datetime.datetime.now()
+        validate_rx_tx_counters(pdp, end_time, start_time)
+
+
+def test_nexthop_group_member_scale(
+    duthost,
+    ptfadapter,
+    localhost,
+    tbinfo,
+    bgp_peers_info,
+    announce_bgp_routes_teardown
+):
+    topo_name = tbinfo['topo']['name']
+    ptf_ip = tbinfo['ptf_ip']
+    pdp = ptfadapter.dataplane
+    bgp_ports = [bgp_info[DUT_PORT] for bgp_info in bgp_peers_info.values()]
+    injection_dut_port = random.choice(bgp_ports)
+    injection_port = [i[PTF_PORT] for i in bgp_peers_info.values() if i[DUT_PORT] == injection_dut_port][0]
+    logger.info("Injection port: %s", injection_port)
+
+    startup_routes = get_all_bgp_ipv6_routes(duthost)
+    ecmp_routes = {r: v for r, v in startup_routes.items() if len(v[0]['nexthops']) == len(bgp_peers_info)}
+    pkts = generate_packets(
+        ecmp_routes,
+        duthost.facts['router_mac'],
+        pdp.get_mac(0, injection_port)
+    )
+    nhipv6 = tbinfo['topo']['properties']['configuration_properties']['common']['nhipv6']
+    picked_prefixes = random.sample(ecmp_routes.keys(), len(ecmp_routes) // 2)
+    picked_routes = {p: ecmp_routes[p] for p in picked_prefixes}
+    routes_to_change = [(r, nhipv6, None) for r in picked_routes]
+
+    bgp_peers = bgp_peers_info.keys()
+    random_half_peers = random.sample(bgp_peers, len(bgp_peers) // 2)
+
+    start_time = datetime.datetime.now()
+    ternimated = Event()
+    # TODO: update device number for multi-servers topo by method port_to_device
+    traffic_thread = Thread(target=send_packets, args=(ternimated, pdp, 0, injection_port, pkts, MAX_PKTS_COUNT))
+    unsafe_flush_counters(pdp)
+    traffic_thread.start()
+    try:
+        change_routes_on_peers(localhost, topo_name, ptf_ip, routes_to_change, ACTION_WITHDRAW, random_half_peers)
+        withdraw_time = datetime.datetime.now()
+        nexthops_to_remove = [b[IPV6_KEY] for n, b in bgp_peers_info.items() if n in random_half_peers]
+        expected_routes = dict(startup_routes)
+        expected_routes.update(
+            remove_nexthops_in_routes(picked_routes, nexthops_to_remove)
+        )
+        while not compare_routes(get_all_bgp_ipv6_routes(duthost), expected_routes):
+            if datetime.datetime.now() - withdraw_time > datetime.timedelta(seconds=MAX_CONVERGENCE_WAIT_TIME):
+                logging.info("Actual routes: %s", get_all_bgp_ipv6_routes(duthost))
+                logging.info("Expected routes: %s", expected_routes)
+                pytest.fail("BGP routes are not stable in long time")
+
+        logger.info("Routes are stable after routes withdrawn: %s", datetime.datetime.now() - withdraw_time)
+
+        ternimated.set()
+        traffic_thread.join()
+        end_time = datetime.datetime.now()
+        validate_rx_tx_counters(pdp, end_time, start_time)
+
+    finally:
```

Comment:
There are some very common patterns:

> preparation
start timer/clear counter/start traffic
> testing
stop traffic/stop timer/validate rxtx

please consider extract this pattern in your next PR, it is very useful